flink-streaming-connectors/flink-connector-kinesis/src/main/java/org/apache/flink/streaming/connectors/kinesis/FlinkKinesisProducer.java

```diff
@@ -160,12 +168,13 @@ public void setCustomPartitioner(KinesisPartitioner<OUT> partitioner) {
 	public void open(Configuration parameters) throws Exception {
 		super.open(parameters);
 
-		KinesisProducerConfiguration config = new KinesisProducerConfiguration();
-		config.setRegion(this.region);
-		config.setCredentialsProvider(new StaticCredentialsProvider(new BasicAWSCredentials(this.accessKey, this.secretKey)));
+		KinesisProducerConfiguration producerConfig = new KinesisProducerConfiguration();
+
+		producerConfig.setRegion(configProps.getProperty(KinesisConfigConstants.CONFIG_AWS_REGION));
+		producerConfig.setCredentialsProvider(AWSUtil.getCredentialsProvider(configProps));
 		//config.setCollectionMaxCount(1);
 		//config.setAggregationMaxCount(1);
```

Comment:
I know this is out of scope for the JIRA, but maybe it would make sense to introduce some producer specific configuration keys for setting relevant producer configuration settings here.

Would you like to add this as part of the PR or do you think we should file a follow up JIRA?